Decide which Eclipse version our eclipsePlugin supports

We had related discussion in #205; It would be nice to stop supporting ancient release like Juno.
Maybe we can drop supports for Mars, Luna, Kepler, Juno and older releases?

- [x] Oxygen (release candidate) (4.7)
- [x] Neon (stable) (4.6)
- [ ] Mars (4.5) which starts new directory layout
- [ ] Luna (4.4)
- [ ] Kepler (4.3)
- [ ] Juno (4.2, 3.8)
- [ ] Previous to Juno (<=4.1)

Note that current SpotBugs Eclipse Plugin cannot be built with Neon.
If we update `.travis.yml` to build with supported Eclipse version, we may need to update our code.

```
\spotbugs\eclipsePlugin\src\de\tobject\findbugs\FindbugsPlugin.java:250: error: incompatible types: String cannot be converted to Plugin
        ResourcesPlugin.getWorkspace().addSaveParticipant(PLUGIN_ID, saveParticipant);
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -21,7 +21,7 @@ before_install:
   - if [[ $JDK_FOR_TEST == "oraclejdk9" ]]; then sudo dpkg -i deps/oracle-java9-installer.deb; fi
 
 install:
-  - if [[ ! -e deps/eclipse.tar.gz ]]; then wget http://www.eclipse.org/downloads/download.php?file=/eclipse/downloads/drops/R-3.6.2-201102101200/eclipse-SDK-3.6.2-linux-gtk-x86_64.tar.gz -O deps/eclipse.tar.gz; fi
+  - if [[ ! -e deps/eclipse.tar.gz ]]; then wget https://www.eclipse.org/downloads/download.php?file=/eclipse/downloads/drops4/R-4.6.3-201703010400/eclipse-SDK-4.6.3-linux-gtk-x86_64.tar.gz&r=1 -O deps/eclipse.tar.gz; fi
   - tar xzvf deps/eclipse.tar.gz eclipse
   - echo eclipseRoot.dir=$(pwd)/eclipse > eclipsePlugin/local.properties
 

